app/Http/Traits/UploadFileTrait.php

```
@@ -2,30 +2,41 @@
-        $oldAttachment = explode("/", $oldFile);
-        $hashmake = Hash::make('secret');
-        $hashname = substr(hash('sha256', $hashmake), 0, 20);
-        $originalName = $file->getClientOriginalName();
-        $file->storePubliclyAs($fileLocation . $hashname, $originalName, 'public');
-        // FILE LOCATION MUST FOLLOW THE SAME STRUCTURE OF public/*/*/*hashedname/*originalname
-        // NOT TESTED MIGHT DELETE ANOTHER FILE
-        Storage::deleteDirectory("public/" . $oldAttachment[0] . "/" . $oldAttachment[1] . "/" . $oldAttachment[2]);
-        return $fileLocation . $hashname . "/" . $originalName;
+        $oldfileUniqueFolder = explode("/", $oldFile);
+        array_pop($oldfileUniqueFolder);
+        Storage::deleteDirectory("public/" . implode("/", $oldfileUniqueFolder)); // DELETE OLD FILE
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\Storage' in method 'replaceUploadFile'. (undefined)

(StaticAccess)